### PR TITLE
update to powsybl dependencies 1.3.0

### PR DESCRIPTION
**Pull request by EtienneLt**
Signed-off-by: Etienne LESOT <etienne.lesot@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

no

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

update powsybl core, powsybl open loadflow and powsybl single linde diagram versions 

**What is the new behavior (if this is a feature change)?**
use powsybl core 4.10, openloadflow 0.23 and powsybl single line diagram 2.13

### CODE REVIEW

File: java/src/main/java/com/powsybl/python/network/NetworkUtil.java [hunk 1]
```diff
@@ -173,28 +173,28 @@ static List<String> getElementsIds(Network network, PyPowsyblApiHeader.ElementTy
     public static Stream<TemporaryLimitData> getLimits(Network network) {
         Stream.Builder<TemporaryLimitData> limits = Stream.builder();
         network.getBranchStream().forEach(branch -> {
-            addLimit(limits, branch,  branch.getCurrentLimits1(), ONE);
-            addLimit(limits, branch,  branch.getCurrentLimits2(), TWO);
-            addLimit(limits, branch,  branch.getActivePowerLimits1(), ONE);
-            addLimit(limits, branch,  branch.getActivePowerLimits2(), TWO);
-            addLimit(limits, branch,  branch.getApparentPowerLimits1(), ONE);
-            addLimit(limits, branch,  branch.getApparentPowerLimits2(), TWO);
+            addLimit(limits, branch, (LoadingLimits) branch.getCurrentLimits1().orElse(null), ONE);
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/python/network/NetworkUtil.java:176]: @sylvlecl @geofjamg I'm not sure of what should be the behavior here

**Comment by obrix** [comment]
I'm trying to use this branch as a base for my next PR that use some 4.10 features but I have some issues. You seem to use a powsybl dependencies 1.3.0 snapshot version, I tried to use one locally where I just bumped core version to 4.10 RC1 but I still have issue building pypowsybl with that. Is there something else to change ? (version of other dependencies in powsybl-dependencies ?)

**Comment by EtienneLt** [comment]
> I'm trying to use this branch as a base for my next PR that use some 4.10 features but I have some issues. You seem to use a powsybl dependencies 1.3.0 snapshot version, I tried to use one locally where I just bumped core version to 4.10 RC1 but I still have issue building pypowsybl with that. Is there something else to change ? (version of other dependencies in powsybl-dependencies ?)

<powsybl-core.version>4.10.0-SNAPSHOT</powsybl-core.version>
<powsybl-open-loadflow.version>0.23.0-SNAPSHOT</powsybl-open-loadflow.version>
<powsybl-single-line-diagram.version>2.13.0-SNAPSHOT</powsybl-single-line-diagram.version>

I think it works with main branches

File: java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java [hunk 2]
```diff
@@ -148,20 +148,35 @@ public Collection<OperationalLimits> getOperationalLimits() {
...
             @Override
-            public CurrentLimits getCurrentLimits() {
+            public Optional<CurrentLimits> getCurrentLimits() {
                 return branch.getCurrentLimits(side);
             }
 
             @Override
-            public ActivePowerLimits getActivePowerLimits() {
+            public CurrentLimits getNullableCurrentLimits() {
+                return branch.getNullableCurrentLimits1();
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java]: Should be 1 or 2 depending on  `side`

File: java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java [hunk 3]
```diff
@@ -148,20 +148,35 @@ public Collection<OperationalLimits> getOperationalLimits() {
...
+
+            @Override
+            public Optional<ActivePowerLimits> getActivePowerLimits() {
                 return branch.getActivePowerLimits(side);
             }
 
             @Override
-            public ApparentPowerLimits getApparentPowerLimits() {
+            public ActivePowerLimits getNullableActivePowerLimits() {
+                return branch.getNullableActivePowerLimits1();
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java]: Should be 1 or 2 depending on  `side`

File: java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java [hunk 4]
```diff
@@ -148,20 +148,35 @@ public Collection<OperationalLimits> getOperationalLimits() {
...
+            }
+
+            @Override
+            public Optional<ApparentPowerLimits> getApparentPowerLimits() {
                 return branch.getApparentPowerLimits(side);
             }
 
+            @Override
+            public ApparentPowerLimits getNullableApparentPowerLimits() {
+                return null;
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/dataframe/network/adders/OperationalLimitsDataframeAdder.java]: should be `getApparentPowerLimits().orElse(null)`

**Review by sylvlecl — CHANGES_REQUESTED** [review]
Minor fixes to do on flows holder (although it has probably no impact), then OK to merge

**Review by sylvlecl — APPROVED** [review]

**Pull request merged**